src/patient-registration/patient-registration.component.tsx

```diff
@@ -402,12 +401,25 @@ export const PatientRegistration: React.FC = () => {
     deletedNames.forEach(async name => {
       await deletePersonName(name.nameUuid, name.personUuid, abortController);
     });
+
+    const getAfterUrl = patientUuid => {
+      if (existingPatient) {
+        const afterEditUrl = new URLSearchParams(search).get('afterUrl');
+        if (afterEditUrl) {
+          return afterEditUrl;
+        }
+      }
+      const urlTemplate = config.registrationSubmission.afterCreateUrl;
+      return urlTemplate.includes('${patientUuid}')
+        ? interpolateString(urlTemplate, { patientUuid: patientUuid })
+        : urlTemplate;
```

Comment:
```suggestion
      return new URLSearchParams(search).get('afterUrl') || interpolateString(config.links.submitButton, { patientUuid: patientUuid });
```

Should be sufficient. Since it's more or less one line, you could even inline it into the `to` param of the `navigate` call below and get rid of this function entirely.

Reply:
Fixed